mlir/utils/jenkins/Jenkinsfile

```diff
@@ -629,25 +632,27 @@ def withHealthyNode(String baseLabel, Closure<?> healthChecks, Closure<?> body,
         blacklist.each { expr.append(' && !').append(it) }
 
         echo "[withHealthyNode] attempt #${attempt}: looking for '${expr}'"
-        node(expr.toString()) {
-            // Retry ONLY the health-check. We don't want to retry the actual stages
-            try {
-                stage("Health checks on ${env.NODE_NAME}") {
-                    healthChecks()
-                    gitHealthCheck()
+        timeout(time: NODE_ALLOCATION_TIMEOUT, unit: 'MINUTES', activity: true) {
```

Comment:
Does it print useful messages on timeout ? 
I suppose it would be generic "timed out due to inactivity". Can it be more descriptive e.g. "timed out because it could not find health node in predefind timeout limit".